src/client/signatures.rs

```diff
@@ -244,52 +275,77 @@ mod tests {
             timestamp: 0,
             signer: "".to_string(),
         };
-        let err = verifier.fetch_certificate_chain(&collection).unwrap_err();
+
+        let dummy_client: Box<dyn Requester + 'static> = Box::new(DummyClient);
+
+        let err = verifier
+            .fetch_certificate_chain(&dummy_client, &collection)
+            .unwrap_err();
         match err {
             SignatureError::MissingSignatureField() => assert!(true),
             e => assert!(false, "Unexpected error type: {:?}", e),
         };
     }
 
     #[test]
-    fn test_bad_x5u_urls() {
+    fn test_bad_x5u_urls_request_fails() {
         let verifier = DummyVerifier {};
 
-        let _ = set_backend(&ReqwestBackend);
-        let mock_server = MockServer::start();
-        let mock_server_address = mock_server.url("/file.pem");
-        let mut pem_mock = mock_server.mock(|when, then| {
-            when.path("/file.pem");
-            then.status(404);
-        });
+        let test_client: Box<dyn Requester + 'static> = Box::new(DummyClient);
```

Comment:
I think a comment here would help to clarify what we are testing here (ie. error handling).

Because the dummy client always fails, even if the URL is valid and reachable. Baiscally, if our test client would distinguish 404 from unreachable URLs (introspecting the host part or something like that), then it would make more sense to use it here instead.

Reply:
Let me restore this test to use Viaduct as it's much clearer expressed that way!